drivers/display/display_renesas_lcdc.c

```diff
@@ -99,26 +99,46 @@ struct display_smartbond_config {
 	enum display_pixel_format pixel_format;
 };
 
-static inline void lcdc_smartbond_pm_policy_state_lock_get(struct display_smartbond_data *data)
+#ifdef CONFIG_PM_DEVICE
+static inline void lcdc_smartbond_pm_prevent_system_sleep(struct display_smartbond_data *data)
 {
-#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_PM_DEVICE_RUNTIME)
 	if (atomic_test_and_set_bit(data->pm_policy_state_flag, 0) == 0) {
 		/*
 		 * Prevent the SoC from etering the normal sleep state as PDC does not support
 		 * waking up the application core following LCDC events.
 		 */
 		pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
 	}
-#endif
 }
 
-static inline void lcdc_smartbond_pm_policy_state_lock_put(struct display_smartbond_data *data)
+static inline void lcdc_smartbond_pm_allow_system_sleep(struct display_smartbond_data *data)
 {
-#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_PM_DEVICE_RUNTIME)
 	if (atomic_test_and_clear_bit(data->pm_policy_state_flag, 0) == 1) {
 		/* Allow the SoC to enter the nornmal sleep state once LCDC is inactive */
 		pm_policy_state_lock_put(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
 	}
+}
+#endif
+
+static inline void lcdc_smartbond_pm_policy_state_lock_get(const struct device *dev)
+{
+#ifdef CONFIG_PM_DEVICE
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+	pm_device_runtime_get(dev);
+#else
+	lcdc_smartbond_pm_prevent_system_sleep(dev->data);
+#endif
+#endif
+}
```

Comment:
I may be missing something, but I think it is mixing things. When you have `PM_DEVICE_RUNTIME` enabled you are just holding a reference to the device while when it is disabled you are basically preventing the SoC to go standby.

Question, does this driver lose context when the SoC is standby ? If so, calling `pm_device_runtime_get` won't prevent it.

Reply:
> Question, does this driver lose context when the SoC is standby ? If so, calling pm_device_runtime_get won't prevent it.

Yes, LCD controller will be powered down when the SoC enters standby because it is in the same power domain. That's why in RUNTIME we are putting / getting constraints in the system to avoid the SoC  to go standby when the device is resumed or suspended.

What we try to achieve, when `PM_DEVICE_RUNTIME` is enabled, is to reduce the power consumption when SoC is active.

SoC is waking up from standby only through the Power Domain Controller (PDC). PDC can wake up the system from specific triggers. If a device doesn't generate a trigger for the PDC, it cannot wake up the SoC.
When SoC is in standby, it is powered down, so even if a device in a different power domain is up when it will create an IRQ signal the SoC will not consume it.